diopter/sanitizer.py

```diff
@@ -363,6 +372,67 @@ def check_for_ub_and_address_sanitizer_errors(
                 print("Sanitizer checks passed")
             return SanitizationResult()
 
+    def check_for_memory_sanitizer_errors(
```

Comment:
Since this is almost a duplicate of the other sanitizer function, maybe it makes sense to use a single one, e.g., `check_for_sanitizer_errors(self, program: SourceProgram, sanitizer_flag: str) -> SanitizationResult:` 

then when actually compiling the program: `  f"-fsanitize=sanitizer_flag"`

and

```
def check_for_memory_sanitizer_errors(self, program: SourceProgram) -> SanitizationResult:
  return self.check_for_sanitizer_errors(program, "memory")
```

same for the other function